### DISCUSSION TITLE
What is the relationship between llama.cpp and llama2.c and llama2.cpp?

### DISCUSSION THREAD

**Discussion by 2catycm** [ggml-org/llama.cpp#3142] [Q&A]
https://github.com/karpathy/llama2.c
https://github.com/leloykun/llama2.cpp

**Comment by 2catycm** [comment]
These two git repo is not based on ggml?

**Comment by monatis** [comment]
No, they are based on GGML. llama2.c is "inspired" by llama.cpp as it's put in its readme and intended to be a very minimalistic ~500-line inference example for further hacking. and llama2.cpp is simply C++ port of that repo.
